src/html-media-source.js

```diff
@@ -159,6 +159,19 @@ export default class HtmlMediaSource extends videojs.EventTarget {
       });
     };
 
+    this.onPlayerSeeking_ = (event) => {
+      this.sourceBuffers.forEach((sourceBuffer) => {
+        if (sourceBuffer.transmuxer_) {
+          let range = this.player_.buffered();
+          let current = this.player_.currentTime();
+
+          if (current >= range.end(0) || current <= range.start(0)) {
```

Comment:
Having contrib-hls handle the message would simplify this alot. I agree we should come up with a solution that can take advantage of the code and information contrib-hls has.

Historically we've tried to keep this project as a shim for MSE so that contrib-hls would not need knowledge of whether it was working with this project or native media source. (Even though contrib-hls is always using our shim). Eventually we would like to do away with this project and pull functionality into contrib-hls so we don't have to deal with these constraints that we keep running into. However, re-examining how the two projects interact, we are already [breaking that principle by calling ](https://github.com/videojs/videojs-contrib-hls/blob/master/src/master-playlist-controller.js#L445)`addSeekableRange_` from contrib-hls. We also communicate `mediachange` between these projects through an [event on the tech](https://github.com/videojs/videojs-contrib-hls/blob/master/src/master-playlist-controller.js#L505) that [contrib-media-sources listens to](https://github.com/videojs/videojs-contrib-media-sources/blob/master/src/html-media-source.js#L189). I don't think we should add new apis for contrib-hls to call (even though we are doing it once). I think the event solution is reasonable and keeps with the spirit of being a native shim. If contrib-hls was working with a native mediasource, triggering an event on the tech would not break anything as there would just be nothing listening to it. 

So instead of doing the reset on seek events, lets listen for a custom event on the player, and do the reset when that event is triggered. Then all that would be needed is for contrib-hls to trigger the event at the right moment